File: Assets/Platform/Project/Game/Base/Lua/View/Common/AlertPanel.lua
AlertPanel = ClassPanel("AlertPanel")
AlertPanel.Instance = nil
AlertPanel.lastData = nil

local this = nil

function AlertPanel:OnInitUI()
    this = self
    local content = self:Find("Content")
    this.msgTxt = content:Find("MsgTxt"):GetComponent(TypeText)
    this.okCenterBtn = content:Find("OkCenterButton").gameObject
    this.okBtn = content:Find("OkButton").gameObject
    this.cancelBtn = content:Find("CancelButton").gameObject
    this.TitleText = content:Find("Background/Title/TitleText")
    this.AddListenerEvent()
end

function AlertPanel:OnOpened(data)
    AlertPanel.Instance = self
    this = self
    this.UpdateData(data)
end

function AlertPanel:OnClosed()
    AlertPanel.Instance = nil
    this.lastData = nil
end

------------------------------------------------------------------
--
function AlertPanel.AddListenerEvent()
    this:AddOnClick(this.okCenterBtn, this.OnOkBtnClick)
    this:AddOnClick(this.okBtn, this.OnOkBtnClick)
    this:AddOnClick(this.cancelBtn, this.OnCancelBtnClick)
end

function AlertPanel.RemoveListenerEvent()

end
------------------------------------------------------------------
--
--关闭
function AlertPanel.ClosePanel()
    if Alert.openPanelConfig == this.panelConfig then
        Alert.openPanelConfig = nil
    end
    Alert.isOpen = false
    this:Close()
end

--================================================================
--更新数据
function AlertPanel.UpdateData(data)
    if data == nil then
        this.ClosePanel()
        return
    end
    if data.level == nil then
        data.level = AlertLevel.Normal
    end

    if this.lastData ~= nil and this.lastData.level > data.level then
        return
    end

    this.lastData = data

    this.msgTxt.text = this.lastData.message
    --LogError("data.title", data.title)
    --UIUtil.SetText(this.TitleText, data.title)

    if this.lastData.type == AlertType.Prompt then
        this.ShowPromptDisplay()
    else
        this.ShowAlertDisplay()
    end
end

------------------------------------------------------------------
--
function AlertPanel.ShowAlertDisplay()
    UIUtil.SetActive(this.okCenterBtn, true)
    UIUtil.SetActive(this.okBtn, false)
    UIUtil.SetActive(this.cancelBtn, false)
end

function AlertPanel.ShowPromptDisplay()
    UIUtil.SetActive(this.okCenterBtn, false)
    UIUtil.SetActive(this.okBtn, true)
    UIUtil.SetActive(this.cancelBtn, true)
end

function AlertPanel.OnOkBtnClick()
    local temp = this.lastData
    this.ClosePanel()
    if temp ~= nil and temp.okCallback ~= nil then
        temp.okCallback()
    end
end

function AlertPanel.OnCancelBtnClick()
    local temp = this.lastData
    this.ClosePanel()
    if temp ~= nil and temp.cancelCallback ~= nil then
        temp.cancelCallback()
    end
end

File: Assets/Platform/Scripts/UI/Panels/AlertPanel.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

public class AlertPanel : MonoBehaviour
{
    Text msgTet;

    Button okCenterBtn;
    Button okBtn;
    Button cancelBtn;

    private Action mOnOkCallback = null;
    private Action mOnCancelCallback = null;
    private AlertLevel mAlertLevel = AlertLevel.Normal;

    private WindowTweener tweener = null;
    private int clickBtnType = 0;

    void Awake()
    {
        msgTet = transform.Find("Content/MsgTxt").GetComponent<Text>();
        okCenterBtn = transform.Find("Content/OkCenterButton").GetComponent<Button>();
        okBtn = transform.Find("Content/OkButton").GetComponent<Button>();
        cancelBtn = transform.Find("Content/CancelButton").GetComponent<Button>();

        tweener = this.GetComponentInChildren<WindowTweener>();
    }

    void Start()
    {
        okBtn.onClick.AddListener(OnOkBtnClick);
        okCenterBtn.onClick.AddListener(OnOkBtnClick);
        cancelBtn.onClick.AddListener(OnCancelBtnClick);
    }

    private void OnOkBtnClick()
    {
        this.clickBtnType = 0;
        this.InternalClose();

    }

    private void OnCancelBtnClick()
    {
        this.clickBtnType = 1;
        this.InternalClose();
    }

    public void Open(string msg, AlertType type, string okBtnTxt, Action onOkCallback, string cancelBtnTxt,
        Action onCancelCallback, AlertLevel level = AlertLevel.Normal)
    {
        SetObjActive(this.gameObject, true);

        if (mAlertLevel > level)
        {
            return;
        }

        this.msgTet.text = msg;
        mAlertLevel = level;
        mOnOkCallback = onOkCallback;
        mOnCancelCallback = onCancelCallback;
        if (type == AlertType.Ok)
        {
            SetObjActive(okCenterBtn.gameObject, true);
            SetObjActive(okBtn.gameObject, false);
            SetObjActive(cancelBtn.gameObject, false);
        }
        else
        {
            SetObjActive(okCenterBtn.gameObject, false);
            SetObjActive(okBtn.gameObject, true);
            SetObjActive(cancelBtn.gameObject, true);
        }
    }

    private void SetObjActive(GameObject go, bool value)
    {
        if (value)
        {
            if (!go.activeSelf)
            {
                go.SetActive(true);
            }
        }
        else
        {
            if (go.activeSelf)
            {
                go.SetActive(false);
            }
        }
    }

    private void Clear()
    {
        msgTet.text = "";
        mOnOkCallback = null;
        mOnCancelCallback = null;
        mAlertLevel = AlertLevel.Normal;
    }

    public void Close()
    {
        SetObjActive(this.gameObject, false);
        Clear();
    }

    private void InternalClose() 
    {
        if (this.tweener != null)
        {
            this.tweener.PlayCloseAnim(this.OnCompleted);
        }
        else 
        {
            this.OnCompleted();
        }
    }

    private void OnCompleted() 
    {
        SetObjActive(this.gameObject, false);
        if (this.clickBtnType == 0)
        {
            if (mOnOkCallback != null)
            {
                mOnOkCallback.Invoke();
            }
        }
        else 
        {
            if (mOnCancelCallback != null)
            {
                mOnCancelCallback.Invoke();
            }
        }
        Clear();
    }
}